pandas/tests/test_common.py

```diff
@@ -7,6 +7,9 @@
 import numpy as np
 import pytest
 
+sys.modules.pop("bz2", None)  # Remove 'bz2' from available modules for testing
+from pandas.compat import get_bz2_file
```

Comment:
This doesn't look reliable to me; It seems better to formulate this test in this style:

https://github.com/pandas-dev/pandas/blob/3860884941fd98839ab6c5260cece486367bc96d/pandas/tests/plotting/test_converter.py#L46-L58

Just import pandas and then assert bz2 is not in sys.modules.

Reply:
Happy to, but would it pose a problem that it doesn't get measured in the test coverage?

Reply:
Just pushed.

Reply:
Thanks! It's a good point about being able to measure test coverage, but I'd rather have reliable tests than a high coverage report on tests that are unreliable :smile:. That said, if the previous way was known to be reliable then I'm fine with it. But you're modifying global state on import (pytest collection, I think?) and I'm not sure what guarantees happen between collection and when the tests start running.